common/src/main/java/io/netty/util/concurrent/AbstractScheduledEventExecutor.java

```diff
@@ -0,0 +1,236 @@
+     */
+    protected final void cancelDelayedTasks() {
+        assert inEventLoop();
+        Queue<ScheduledFutureTask<?>> delayedTaskQueue = scheduledTaskQueue;
+        if (delayedTaskQueue == null || delayedTaskQueue.isEmpty()) {
+            return;
+        }
+
+        final ScheduledFutureTask<?>[] delayedTasks =
+                delayedTaskQueue.toArray(new ScheduledFutureTask<?>[delayedTaskQueue.size()]);
+
+        for (ScheduledFutureTask<?> task: delayedTasks) {
+            task.cancel(false);
+        }
+
+        delayedTaskQueue.clear();
+    }
+
+    /**
+     * @see {@link #pollScheduledTask(long)}
+     */
+    protected final Runnable pollScheduledTask() {
+        return pollScheduledTask(nanoTime());
+    }
+
+    /**
+     * Return the {@link Runnable} which is ready to be executed with the given {@code nanoTime}.
+     * You should use {@link #nanoTime()} to retrieve the the correct {@code nanoTime}.
+     */
+    protected final Runnable pollScheduledTask(long nanoTime) {
+        assert inEventLoop();
+
+        Queue<ScheduledFutureTask<?>> delayedTaskQueue = scheduledTaskQueue;
+        ScheduledFutureTask<?> delayedTask = delayedTaskQueue == null ? null : delayedTaskQueue.peek();
+        if (delayedTask == null) {
+            return null;
+        }
+
+        if (delayedTask.deadlineNanos() <= nanoTime) {
+            delayedTaskQueue.remove();
+            return delayedTask;
+        }
+        return null;
+    }
+
+    /**
+     * Return the nanoseconds when the next scheduled task is ready to be run or {@code -1} if no task is scheduled.
+     */
+    protected final long nextScheduledTaskNano() {
+        assert checkInEventLoop();
+
+        Queue<ScheduledFutureTask<?>> delayedTaskQueue = scheduledTaskQueue;
+        ScheduledFutureTask<?> delayedTask = delayedTaskQueue == null ? null : delayedTaskQueue.peek();
+        if (delayedTask == null) {
+            return -1;
+        }
+        return Math.max(0, delayedTask.deadlineNanos() - nanoTime());
+    }
+
+    final ScheduledFutureTask<?> peekScheduledTask() {
+        assert checkInEventLoop();
+
+        Queue<ScheduledFutureTask<?>> delayedTaskQueue = scheduledTaskQueue;
+        if (delayedTaskQueue == null) {
+            return null;
+        }
+        return delayedTaskQueue.peek();
+    }
+
+    /**
+     * Returns {@code true} if a scheduled task is ready for processing.
+     */
+    protected final boolean hasScheduledTasks() {
+        assert checkInEventLoop();
+
+        Queue<ScheduledFutureTask<?>> delayedTaskQueue = scheduledTaskQueue;
+        ScheduledFutureTask<?> delayedTask = delayedTaskQueue == null ? null : delayedTaskQueue.peek();
+        return delayedTask != null && delayedTask.deadlineNanos() <= nanoTime();
+    }
+
+    @Override
+    public  ScheduledFuture<?> schedule(Runnable command, long delay, TimeUnit unit) {
+        if (command == null) {
```

Comment:
Consider using [checkNotNull](https://github.com/netty/netty/blob/master/common/src/main/java/io/netty/util/internal/ObjectUtil.java#L29) to simplify a bit.